### ISSUE TITLE
Website quickstart examples need to be checked

### ISSUE THREAD

**Issue by edwardsmith999** [Crompulence/cpl-library#25]
Issues with Demo: a minimum send/recv example on ARCHER2 identified by Gavin Pringle (https://www.research.ed.ac.uk/en/persons/gavin-pringle)

**Comment by gavinpringle** [comment]
I took the fortran code from the Quick Start Guide "Demo: a minimum send/recv example". The example fortran files are not in the git repo under examples. The code text does not cut-n-paste directly into a bash ssh session. I pasted it into notepad++, but found TABS, which I removed.  I could then cut-n-paste from notepad++ into a bash window just fine, and got it to compile.    

Running with two MPI_COMM_WORLDs hangs, and running with a single MPI_COMM_WORLD throws an error.

First, the output of the hanging run:

 Only MD realm present in MPI_COMM_WORLD
 Cannot find port file: ./port. Waiting...
 Cannot find port file: ./port. Waiting...
 Cannot find port file: ./port. Waiting...
 Cannot find port file: ./port. Waiting...
 Only CFD realm present in MPI_COMM_WORLD
 opened port: tag#0$connentry#0200BEC60AFD1D960000000000000000$ Attempting to write to file:./port
 Portname written to file ./port
 connection accepted to root of           16  procs.
 connection accepted to root of           16  procs.
 connection accepted to root of           16  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 accepted connection on port to root of            4  procs.
 connection accepted to root of           16  procs.
 Rank on realm            2  is            3  of            4   and rank on intercomm is           18  of           20
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:          18
 Rank on realm            2  is            4  of            4   and rank on intercomm is           19  of           20
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:          19
 Rank on realm            1  is            1  of           16   and rank on intercomm is            0  of           20
 Rank on realm            2  is            2  of            4   and rank on intercomm is           17  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           0
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:          17
 Rank on realm            1  is            2  of           16   and rank on intercomm is            1  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           1
 Rank on realm            1  is            3  of           16   and rank on intercomm is            2  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           2
 Rank on realm            1  is            4  of           16   and rank on intercomm is            3  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           3
 Rank on realm            1  is            5  of           16   and rank on intercomm is            4  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           4
 Rank on realm            1  is            6  of           16   and rank on intercomm is            5  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           5
 Rank on realm            1  is            7  of           16   and rank on intercomm is            6  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           6
 Rank on realm            1  is            8  of           16   and rank on intercomm is            7  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           7
 Rank on realm            1  is           13  of           16   and rank on intercomm is           12  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          12
 Rank on realm            1  is           14  of           16   and rank on intercomm is           13  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          13
 Rank on realm            1  is           15  of           16   and rank on intercomm is           14  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          14
 Rank on realm            1  is           16  of           16   and rank on intercomm is           15  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          15
 Rank on realm            1  is           11  of           16   and rank on intercomm is           10  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          10
 Rank on realm            2  is            1  of            4   and rank on intercomm is           16  of           20
 Rank on realm            1  is           12  of           16   and rank on intercomm is           11  of           20
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:          16
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          11
 Rank on realm            1  is           10  of           16   and rank on intercomm is            9  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           9
 Rank on realm            1  is            9  of           16   and rank on intercomm is            8  of           20
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           8

Second, the output of the run that throws an error:


 MPMD mode, CFD and MD both share MPI_COMM_WORLD
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:           0
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:           1
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:           2
 Completed CPL communicator init for MD  , CPL_WORLD_COMM ID:           3

   ________/\\\\\\\\\__/\\\\\\\\\\\\\____/\\\_____________
    _____/\\\////////__\/\\\/////////\\\_\/\\\_____________
     ___/\\\/___________\/\\\_______\/\\\_\/\\\_____________
      __/\\\_____________\/\\\\\\\\\\\\\/__\/\\\_____________
       _\/\\\_____________\/\\\/////////____\/\\\_____________
        _\//\\\____________\/\\\_____________\/\\\_____________
         __\///\\\__________\/\\\_____________\/\\\_____________
          ____\////\\\\\\\\\_\/\\\_____________\/\\\\\\\\\\\\\\\_
           _______\/////////__\///______________\///////////////__

                      C P L  -  L I B R A R Y

Non-coherent number of processes in MD      4 no equal to      4 X      2 X      2
Non-coherent number of processes in MD      4 no equal to      4 X      2 X      2
Non-coherent number of processes in MD      4 no equal to      4 X      2 X      2
Non-coherent number of processes in MD      4 no equal to      4 X      2 X      2
MPICH ERROR [Rank 0] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001230] - Abort(0) (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 0) - process 0

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 0) - process 0
MPICH ERROR [Rank 1] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001230] - Abort(0) (rank 1 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 0) - process 1

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 0) - process 1
MPICH ERROR [Rank 2] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001230] - Abort(0) (rank 2 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 0) - process 2

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 0) - process 2
MPICH ERROR [Rank 3] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001230] - Abort(0) (rank 3 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 0) - process 3

 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           4
aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 0) - process 3
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           8
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           9
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          10
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          14
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          15
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          17
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          18
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          19
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           5
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           6
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:           7
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          11
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          12
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          13
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
 Completed CPL communicator init for CFD , CPL_WORLD_COMM ID:          16
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
Non-coherent number of processes in CFD     16 no equal to      2 X      2 X      1
MPICH ERROR [Rank 4] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 4 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 4

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 4
MPICH ERROR [Rank 6] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 6 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 6

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 6
MPICH ERROR [Rank 7] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 7 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 7

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 7
MPICH ERROR [Rank 8] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 8 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 8

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 8
MPICH ERROR [Rank 9] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 9 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 9

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 9
MPICH ERROR [Rank 11] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 11 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 11

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 11
MPICH ERROR [Rank 12] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 12 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 12

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 12
MPICH ERROR [Rank 13] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 13 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 13

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 13
MPICH ERROR [Rank 14] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 14 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 14

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 14
MPICH ERROR [Rank 15] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 15 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 15

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 15
MPICH ERROR [Rank 16] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 16 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 16

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 16
MPICH ERROR [Rank 19] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 19 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 19

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 19
MPICH ERROR [Rank 5] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 5 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 5

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 5
MPICH ERROR [Rank 10] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 10 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 10

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 10
MPICH ERROR [Rank 17] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 17 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 17

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 17
MPICH ERROR [Rank 18] [job id 1833633.0] [Thu Jun 16 15:44:32 2022] [nid001238] - Abort(1) (rank 18 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 18

aborting job:
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 18
srun: error: nid001230: tasks 0-3: Exited with exit code 255
srun: launch/slurm: _step_signal: Terminating StepId=1833633.0+0
srun: launch/slurm: _step_signal: Terminating StepId=1833633.0+1
srun: error: nid001238: tasks 0-15: Exited with exit code 255
